src/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/FileBasedPrograms/FileBasedProgramsProjectSystem.cs

```diff
@@ -50,17 +53,34 @@ public FileBasedProgramsProjectSystem(
         _lspServices = lspServices;
         _logger = loggerFactory.CreateLogger<FileBasedProgramsProjectSystem>();
         _projectXmlProvider = projectXmlProvider;
+        // Lazily create the canonical misc files loader
+        _canonicalMiscFilesLoader = new Lazy<CanonicalMiscFilesProjectLoader>(() =>
```

Comment:
I find this somewhat suspect, it doesn't seem like one project system should "contain" another project system.

Reply:
This is the last bit I am not quite satisfied with. It's possible it is fine, I just need to put a little more scrutiny on it. I was wondering, is the Lazy partially functioning to break a cycle during initialization?

Reply:
No - it can very well not be lazy now actually.  Artifact of when the ctor actually did something.  I will make it not lazy, but not sure that covers everything you were thinking about.

Reply:
I think generally the right path here is to allow multiple misc files providers.   The LSPWorkspaceManager would be responsible for iterating them to see which one could handle the document (likely ordered, with FBP first, then canonical).  

Then the FBP provider here doesn't need to know about the canonical provider.

It's a little larger change though and I'd prefer to do in a followup if you are OK with that.

Reply:
The state management of all the different ways a file can be misc over time, is pretty hard, and severely under-tested. It will be great to be able to put more attention into that area.

I think just getting unstuck on bringing in more test coverage, how to represent more complex scenarios in the test harness, etc, will do us an immense amount of good.